### ISSUE TITLE
Service catalog helm chart instal fails on kubernetes 1.16

### ISSUE THREAD

**Issue by subhankarc** [kubernetes-sigs/service-catalog#2716]
# Bug Report 

**What happened**:
```
helm repo add svc-cat "https://svc-catalog-charts.storage.googleapis.com"
helm repo update
helm install svc-cat/catalog --name catalog --namespace catalog
```

fails on 1.16.0 kubernetes version with error `Error: validation failed: unable to recognize "": no matches for kind "Deployment" in version "extensions/v1beta1"`

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**: NA

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16.0
- service-catalog version: 0.2.1 and 0.3.0-beta.0
- Cloud provider or hardware configuration:
- Do you have api aggregation enabled? 
  - Do you see the configmap in kube-system? 
  - Does it have all the necessary fields?
    - `kubectl get cm -n kube-system extension-apiserver-authentication -o yaml` and look for `requestheader-XXX` fields
- Install tools:
  - Did you use helm? What were the helm arguments? Did you `--set` any extra values?
- Are you trying to use ALPHA features? Did you enable them?

**Comment by mszostok** [comment]
That's is true because in Kubernetes 1.16 the  **extensions/v1beta1** version for  **Deployment** kind was removed. We not yet switched to this version. 

source: https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/


But we will do that in the upcoming two weeks.

**Comment by mszostok** [comment]
@subhankarc are u interested to provide pull-request for that change?

**Comment by subhankarc** [comment]
Hi @mszostok 

Sure, but I am on vacation till 9th, will do it after that.

**Comment by kyschouv** [comment]
Is this new helm chart version going to be published at some point? 1.16 is hitting Azure now, and we can't install service catalog on it (unless we pull this repo).
